OsmAnd-telegram/src/net/osmand/telegram/helpers/TelegramHelper.kt

```diff
@@ -653,28 +637,63 @@ class TelegramHelper private constructor() {
 		}
 	}
 
-	private fun sendLiveLocationImpl(chatLivePeriods: Map<Long, Long>, latitude: Double, longitude: Double) {
-		val location = TdApi.Location(latitude, longitude)
-		chatLivePeriods.forEach { (chatId, livePeriod) ->
-			synchronized(pausedLiveChatIds) {
-				if (pausedLiveChatIds.contains(chatId)) {
-					return@forEach
+	private fun recreateLiveLocationMessage(chatId: Long, msgId: Long, content: TdApi.InputMessageLocation) {
+		if (msgId != -1L) {
+			log.info("recreateLiveLocationMessage - $msgId")
+			val array = LongArray(1)
+			array[0] = msgId
+			client?.send(TdApi.DeleteMessages(chatId, array, true)) { obj ->
+				when (obj.constructor) {
+					TdApi.Ok.CONSTRUCTOR -> {
+						sendNewLiveLocationMessage(chatId, content)
```

Comment:
What if delete message will not run callback? Will new Location message be sent anyway?

Reply:
yes, new location message will be sent anyway in sendLiveLocationImpl, because of current message id made to  -1 before recreating the new message. I made a similar setting to -1 in onDeleteMessages if this callback fails or if a user deletes a message from the telegram

Reply:
onDeleteMessages deletes only if currentMessageId among deleted messages